### PR TITLE
Conda publish

### PR DESCRIPTION
**Pull request by benoit-cty**
#### Technical changes

- Publish to conda
- Test that the package works in Windows
- Fix duplicate CI runs (not related to conda).
- Add documentation on how to use with Conda in README.md
- Add documentation on how to publish to Conda in .conda/README.md

### CODE REVIEW

File: .conda/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+The CI automaticaly upload the PyPi package, see the `.github/workflow.yml`, step `publish-to-conda`.
+
+## Manual actions made to make it works the first time
+
+- Create an account on https://anaconda.org.
+- Create a token on https://anaconda.org/openfisca/settings/access with _Allow write access to the API site_. Warning, it expire on 2023/01/13.
+- Put the token in a CI env variable ANACONDA_TOKEN.
+
+## Manual actions before CI exist
```
Comment (sandcha) [inline .conda/README.md:19]: exists? Or this suggestion:
```suggestion
## Manual actions before CI configuration

```

File: .conda/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+about:
+  home: https://openfisca.org
+  license_family: AGPL
+  license: AGPL-3.0-only
+  license_file: LICENSE
+  summary: "A versatile microsimulation free software"
+  doc_url: https://openfisca.org
+  dev_url: https://github.com/openfisca/openfisca-core/
+  description: This package contains the core features of OpenFisca, which are meant to be used by country packages such as OpenFisca-France.
```
Comment (sandcha) [inline .conda/meta.yaml:100]: > Use an international example?
```suggestion
  description: This package contains the core features of OpenFisca, which are meant to be used by country packages such as OpenFisca-Country-Template.
```

File: .conda/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: PYPI_URL
+  sha256: PYPI_SHA256
+
+build:
+  noarch: python
+  number: 4
```
Comment (sandcha) [inline .conda/meta.yaml:20]: What's the meaning of `number` and `4` here? 😊

File: .github/get_pypi_info.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import argparse
```
Comment (sandcha) [inline .github/get_pypi_info.py:1]: > 👍 Tested in https://github.com/openfisca/openfisca-france/pull/1786; same script.

**Review by sandcha — CHANGES_REQUESTED** [review]
Thank you @benoit-cty for this update and all the documentation!
In this review, you will find
* a question about an error on dev dependencies install
* and some suggestion on the order of the installation steps to guide new users.

File: .conda/meta.yaml (hunk 3 above)
Comment (benoit-cty) [reply .conda/meta.yaml:20]: `number: 4` is a sub-version to allow publishing the same PyPi version multiple time.
It has to be 0 when we switch to a new version.
I will do it.

**Comment by benoit-cty** [comment]
@sandcha : I've done all the changes.

**Comment by MattiSG** [comment]
I'm a bit surprised that this PR has not been merged while Core is seemingly [already published](https://anaconda.org/conda-forge/openfisca-core) 🤔 😅

**Comment by benoit-cty** [comment]
> I'm a bit surprised that this PR has not been merged while Core is seemingly [already published](https://anaconda.org/conda-forge/openfisca-core) thinking sweat_smile
Hi Matti,
It's because Core has it's own FeedStock that create a PR in https://github.com/conda-forge/openfisca-core-feedstock/pulls when a new package is detected on PyPi to publish it to channel `conda-forge`.

This PR is for automatically publish to channel `openfisca` in CI and document all the process.

File: test_dpath.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import dpath.util
```
Comment (sandcha) [inline test_dpath.py:1]: @benoit-cty Is this file here to test that dpath works as expected on conda install or is it a residue from another subject? It's in the root directory and I don't remember seeing it on this PR on first review. Remove it? 🤔

File: README.md [hunk 6]
```diff
@@ -20,7 +23,24 @@ OpenFisca also relies strongly on NumPy. Last four minor versions should work, b
...
 
-If you're developing your own country package, you don't need to explicitly install OpenFisca-Core. It just needs to appear [in your package dependencies](https://github.com/openfisca/openfisca-france/blob/18.2.1/setup.py#L53).
+If you're developing your own country package, you don't need to explicitly install OpenFisca-Core. It just needs to appear [in your package dependencies](https://github.com/openfisca/openfisca-france/blob/100.0.0/setup.py#L60).
+
+Please note that since version [35.7.7](https://anaconda.org/conda-forge/openfisca-core), you could use conda to install OpenFisca-Core with one of this command to run in `Anaconda Powershell Prompt` :
+- `conda create --name openfisca python=3.7` to create an openfisca environment.
+- `conda activate openfisca` to use your new environment.
+- `conda install -c conda-forge openfisca-core` for default dependencies.
+- `conda install -c conda-forge openfisca-core-api` if you want the API part.
+- `conda install -c conda-forge openfisca-core-dev` if you want all the dependencies needed to contribute to the project.
```
[the remark this answers is not in the document]
Comment (sandcha) [reply README.md]: Thank you @benoit-cty! It worked; this last commit (acfcf6566c778a5c69dc12ad6c3a3b90ec6d6ed2) resolved the issue 🙂

File: test_dpath.py (hunk 5 above)
Comment (benoit-cty) [reply test_dpath.py:1]: Nice catch ! I will remove it.

**Review by sandcha — APPROVED** [review]
Thanks a lot @benoit-cty for this new conda packaging and publishing! 🙌

**Comment by sandcha** [comment]
Running `python3 .github/get_pypi_info.py -p OpenFisca-Core` to check why conda build failed generates this meta.yaml where the comment is already updated as in this (local) example:
```yaml
###############################################################################
## File for Anaconda.org
## WARNING : strings 35.8.1, https://files.pythonhosted.org/packages/d9/00/dd774c31b43daca0c844e21f7b56389913996bf7fe3f5e3f2368ec78fd9e/OpenFisca-Core-35.8.1.tar.gz, b843c814015cc5a8e345f342b56906ba2c986d85270acecef9e32a3aea09ef35 and     - dpath <3.0.0,>=1.5.0
    - nptyping ==1.4.4
    - numexpr <=3.0,>=2.7.0
    - numpy <1.21,>=1.11
    - psutil <6.0.0,>=5.4.7
    - pytest <6.0.0,>=4.4.1
    - PyYAML >=3.10
    - sortedcontainers ==2.2.2
    - typing-extensions ==3.10.0.2
 will be replaced
## by CI script get_pypi_info.py, leave them as is.
###############################################################################
...
```

**Pull request merged**